libraries/joomla/table/table.php

```diff
@@ -1382,6 +1501,14 @@ public function reorder($where = '')
 			$query->where($where);
 		}
 
+		// Pre-processing by observers
+		$event = AbstractEvent::create('onTableBeforeReorder', [
+			'subject'	=> &$this,
+			'query'		=> &$query,
+			'where'		=> $where,
+		]);
+		$this->getDispatcher()->dispatch('onTableBeforeReorder', $event);
+
```

Comment:
Just picking this as a representant for all the others:
`$this` is always propagated as a reference, since it is an object. So `'subject' => $this` is sufficient. Same applies to `'query' => $query`.